### PR TITLE
Drop unused get_shipping_price_estimate function

### PR DESCRIPTION
**Pull request by maarcingebala**
This PR drops unused function: `get_shipping_price_estimate`. As a result, the `apply_taxes_to_shipping_price_range` plugin hook gets removed as it is no longer used.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

**Review by d-wysocki — APPROVED** [review]

File: saleor/plugins/base_plugin.py [hunk 1]
```diff
@@ -237,15 +237,6 @@ def show_taxes_on_storefront(self, previous_value: bool) -> bool:
         """
         return NotImplemented
 
-    def apply_taxes_to_shipping_price_range(
```
Comment (fowczarek) [inline saleor/plugins/base_plugin.py:240]: Should we add this to `breaking changes`?

**Review by fowczarek — APPROVED** [review]

File: saleor/plugins/base_plugin.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/plugins/base_plugin.py:240]: When 3.0 is released, probably yes. Now there is no "breaking changes" section in the changelog for 3.0. I can also create it.

Comment (patrys) [reply saleor/plugins/base_plugin.py:240]: We should keep one updated, so we don't have to reconstruct it from memory at launch.

Comment (maarcingebala) [reply saleor/plugins/base_plugin.py:240]: Added "Braking" section in the changelog.

**Review by korycins — APPROVED** [review]

**Pull request merged**